text/006-bft-based-ordering-service.md

```diff
@@ -0,0 +1,404 @@
+
+# The SmartBFT ordering service
+
+The SmartBFT consensus library has been integrated into a fork of Fabric which was branched from Fabric version 1.4.x and has been tested and benchmarked.
+Later on, it was ported to [Fabric 2.3](https://github.com/SmartBFT-Go/fabric/tree/release-2.3-bft).
+
+## Integrating the SmartBFT consensus library into a Fabric orderer
+
+**High level architecture and component interactions**: In its essence, Fabric expects two things from a consensus library: (a) Receiving transactions, and (b) returning them as blocks. 
+The Fabric component that takes care of both aspects is termed a [Chain](https://github.com/hyperledger/fabric/blob/main/orderer/consensus/consensus.go#L58-L92).
+Each Fabric channel induces its own ledger, and has an instance of a chain that is associated with the bespoken channel.
+
+
+The current Fabric code architecture supports various implementations of consensus protocols, and as such, it needs a uniform API of creating different types of chains. 
+Fabric achieves uniformity in chain creation via an abstraction that creates chains called a [Consenter](https://github.com/hyperledger/fabric/blob/main/orderer/consensus/consensus.go#L19-L27).
+A Consenter receives as input an [aggregation of abilities](https://github.com/hyperledger/fabric/blob/main/orderer/consensus/consensus.go#L97-L140) also known as a "Consenter support" that the Fabric ordering node can provide to a consensus protocol, such as cryptographic signatures, ledger access, etc. and in turn,
+and uses a subset of these abilities to create an instance of the consensus protocol, disguised as a chain.
+
+As the [ordering service framework RFC](https://github.com/hyperledger/fabric-rfcs/pull/50) states, this approach is suboptimal as it forces the [ConsenterSupport](https://github.com/hyperledger/fabric/blob/main/orderer/consensus/consensus.go#L97-L140) interface  
+to evolve as new consensus protocols are added into Fabric.
+
+ 
+To that end, a new type of ordering service node will be created, one that can only run a SmartBFT consensus protocol. 
+Similarly to the existing ordering service, the new ordering service node will posses a chain object per channel.
+However, it will not use the consenter approach, but rather it would initialize its chains directly as a result of an invocation of the channel participation API. 
+The latter exposes a callback that is invoked when joining a channel, and that callback will be implemented by a component that will fill a role similar to the [registrar](https://github.com/hyperledger/fabric/blob/main/orderer/common/multichannel/registrar.go).
+
+
+
+## Implementing the dependencies required by the library
+
+In order to integrate the library, the orderer must implement all needed [dependencies](https://smartbft-go.github.io/godoc/pkg/github.com/SmartBFT-Go/consensus/pkg/api/index.html) required by the library.
+Below we enumerate how the dependencies were realized in the SmartBFT orderer:
```

Comment:
I'm getting confused by the context switching between 'old' orderer approach (v2.x), and the 'new' orderer approach (v3.x). The last section talked about the 'old' approach and then the 'new' approach. Is this section talking about the 'old' orderer approach. Or are these the same across the 'old' and 'new' approaches?

Reply:
This section talks about how the library is instantiated and integrated. 
Integrating the library is the same, of course. 
What differs is what we do in the Fabric side, and it's orthogonal to the library.